src/shared/components/axisTicks/TimeTickInput.tsx

```diff
@@ -0,0 +1,236 @@
+// Libraries
+import React, {
+  ChangeEvent,
+  createRef,
+  CSSProperties,
+  FC,
+  RefObject,
+  useState,
+} from 'react'
+
+// Utils
+import {useOneWayState} from 'src/shared/utils/useOneWayState'
+import {convertUserInputValueToNumOrNaN} from 'src/shared/utils/convertUserInput'
+
+// Components
+import DatePicker from 'src/shared/components/dateRangePicker/DatePicker'
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+import {
+  Appearance,
+  Button,
+  ButtonRef,
+  Columns,
+  ComponentColor,
+  ComponentSize,
+  ComponentStatus,
+  Form,
+  Grid,
+  Input,
+  IconFont,
+  Popover,
+  PopoverInteraction,
+  PopoverPosition,
+} from '@influxdata/clockface'
+
+interface TimeTickInputProps {
+  axisName: string
+  tickOptions: string[]
+  initialTickOptionValue: number | string
+  label: string
+  placeholder?: string
+  setOptions: (optionName: string, arg: string[] | number) => void
+}
+
+export const TimeTickInput: FC<TimeTickInputProps> = props => {
+  const {
+    axisName,
+    tickOptions,
+    initialTickOptionValue,
+    label,
+    placeholder = 'RFC3339',
+    setOptions,
+  } = props
+
+  const [tickOptionInput, setTickOptionInput] = useOneWayState(
+    initialTickOptionValue === initialTickOptionValue
+      ? initialTickOptionValue
+      : ''
+  )
+
+  const [tickOptionInputStatus, setTickOptionInputStatus] = useState<
+    ComponentStatus
+  >(ComponentStatus.Default)
+
+  const [isDatePickerOpen, setIsDatePickerOpen] = useState<boolean>(false)
+
+  const [
+    isOnClickOutsideHandlerActive,
+    setIsOnClickOutsideHandlerActive,
+  ] = useState<boolean>(true)
+
+  const triggerRef: RefObject<ButtonRef> = createRef()
+
+  const updateTickOption = (value?: string | number) => {
+    const convertedValue = convertUserInputValueToNumOrNaN(
+      value === undefined ? tickOptionInput : value
+    )
+    const tickOptionNameWithoutAxis = label.split(' ').join('')
+    const tickOptionNameWithAxis = `${axisName.toLowerCase()}${tickOptionNameWithoutAxis}`
+    const filteredTickOptions = Array.isArray(tickOptions)
+      ? tickOptions.filter(option => option !== tickOptionNameWithAxis)
+      : []
+    setOptions(tickOptionNameWithoutAxis, convertedValue)
+    if (convertedValue === convertedValue) {
+      setOptions('GenerateAxisTicks', [
+        ...filteredTickOptions,
+        tickOptionNameWithAxis,
+      ])
+    } else {
+      setOptions('GenerateAxisTicks', filteredTickOptions)
+      if (tickOptionInput !== '') {
+        setTickOptionInputStatus(ComponentStatus.Error)
+      }
+    }
+  }
+
+  const handleInput = (event: ChangeEvent<HTMLInputElement>) => {
+    setTickOptionInput(event.target.value)
+    setTickOptionInputStatus(ComponentStatus.Default)
+  }
+
+  const handleBlur = () => updateTickOption()
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
```

Comment:
Maybe.

Here, we need it because we want to wait until the user types the whole input before attempting to generate the ticks. Without this, as soon as the user types `1` in Tick Step, it would attempt to generate ticks on a step of 1 millisecond, for example. This helped catch a performance issue in giraffe. But we also do not want eager input fields.

So, a long way of saying: yes, in certain cases, we want this.